### PR TITLE
Fix sprite frame editor crash with null `frames` pointer

### LINKED ISSUE #93845: Editor crashes on undos when accidentally duplicate scene node while editing SpriteFrames.
### Tested versions

Reproducible in v4.2.2.stable.official [15073afe3], v4.3.beta2.official [b75f0485b], and v4.3.beta.custom_build [4ab8fb809].
### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 Ti (NVIDIA; 31.0.15.3623) - AMD Ryzen 5 5600X 6-Core Processor (12 Threads)

### Issue description

When editing SpriteFrames, move the mouse to the main panel and the focus changed to Scene Tree. Use shortcut to duplicate the nodes in the scene tree. After that, try to undo twice. The editor will crash.

https://github.com/godotengine/godot/assets/63407648/e29f41ae-399a-441a-941b-f313bf334c1e

### Steps to reproduce

1. Create Node2D Scene.
2. Add AnimatedSprite2D and create SpriteFrames.
3. Open the SpriteFrames editor, and add a new animation.
4. Move the mouse to the main viewport (no need to click) and use `ctrl` + `d`. 
5. Press `ctrl` + `z` twice.
6. Wait a while, the editor will crash.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/93845

### CODE REVIEW

**Comment by EAinsley** [comment]
This will solve the crash problem, but the error 'editor\editor_node.cpp:7751 - Condition "plugins_list.has(p_plugin)" is true.' still exists. Maybe I should open a separated issue for the plugin_list error?

**Comment by jsjtxietian** [comment]
IMHO this may be a different issue or not, so we can first investigate a little more.

File: editor/plugins/sprite_frames_editor_plugin.cpp [hunk 1]
```diff
@@ -953,7 +953,7 @@ void SpriteFramesEditor::_sync_animation() {
 }
 
 void SpriteFramesEditor::_select_animation(const String &p_name, bool p_update_node) {
-	if (!frames->has_animation(p_name)) {
+	if (frames == nullptr || !frames->has_animation(p_name)) {
```
Comment (KoBeWi) [inline editor/plugins/sprite_frames_editor_plugin.cpp:956]:
```suggestion
	if (frames.is_null() || !frames->has_animation(p_name)) {
```

**Review by KoBeWi — APPROVED** [review]

File: editor/plugins/sprite_frames_editor_plugin.cpp (hunk 1 above)
Comment (jsjtxietian) [reply editor/plugins/sprite_frames_editor_plugin.cpp:956]: Thanks!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**